src/core/operations/BitwiseOp.js

```diff
@@ -45,8 +44,13 @@ const BitwiseOp = {
                     case "Output differential":
                         key[i % key.length] = o;
                         break;
+                    case "Cascade":
+                        k ^= input[i + 1];
```

Comment:
This doesn't quite provide a correct Cascade XOR. The key should be set to the next input byte rather than XOR'd with it. The structure of the code here could also be a bit cleaner so that the function isn't called twice. Try this:

``` javascript
    /**
     * Runs bitwise operations across the input data.
     *
     * @private
     * @param {byteArray} input
     * @param {byteArray} key
     * @param {function} func - The bitwise calculation to carry out
     * @param {boolean} nullPreserving
     * @param {string} [scheme="Standard"]
     * @returns {byteArray}
     */
    _bitOp: function (input, key, func, nullPreserving, scheme="Standard") {
        if (!key || !key.length) key = [0];
        let result = [],
            x = null,
            k = null,
            o = null;

        for (let i = 0; i < input.length; i++) {
            k = key[i % key.length];
            if (scheme === "Cascade") k = input[i + 1] || 0;
            o = input[i];
            x = nullPreserving && (o === 0 || o === k) ? o : func(o, k);
            result.push(x);
            if (scheme !== "Standard" &&
                !(nullPreserving && (o === 0 || o === k))) {
                switch (scheme) {
                    case "Input differential":
                        key[i % key.length] = x;
                        break;
                    case "Output differential":
                        key[i % key.length] = o;
                        break;
                }
            }
        }

        return result;
    },
```

Reply:
Oh, in my testing the output matched that of the Python code provided. I agree the code could be made more concise.

I will make the necessary changes and commit those shortly.